components/client/src/main/java/com/hotels/styx/client/OriginsInventory.java

```diff
@@ -532,93 +501,28 @@ public Builder eventBus(EventBus eventBus) {
             return this;
         }
 
-        public Builder clientWorkerThreadsCount(int clientWorkerThreadsCount) {
-            this.clientWorkerThreadsCount = clientWorkerThreadsCount;
+        // TODO: Mikko: seems bit pointless?
+        public Builder initialOrigins(Set<Origin> origins) {
+            this.initialOrigins = ImmutableSet.copyOf(origins);
             return this;
         }
 
-        public Builder(BackendService backendService) {
-            this.backendService = backendService;
-        }
-
-        public Builder originStatsFactory(OriginStatsFactory originStatsFactory) {
-            this.originStatsFactory = originStatsFactory;
-            return this;
+        public Builder(Id appId) {
+            this.appId = requireNonNull(appId);
         }
 
         public OriginsInventory build() {
             if (metricsRegistry == null) {
                 metricsRegistry = new CodaHaleMetricRegistry();
             }
 
+            await(originHealthMonitor.start());
 
-            healthStatusMonitor = Optional.ofNullable(originHealthStatusMonitorFactory)
-                    .orElseGet(OriginHealthStatusMonitorFactory::new)
-                    .create(backendService.id(), backendService.healthCheckConfig(), () -> originHealthCheckFunction(metricsRegistry));
-
-            return originsInventory(healthStatusMonitor, httpConfigBuilder.build(), metricsRegistry, originStatsFactory);
-        }
-
-        private OriginsInventory originsInventory(OriginHealthStatusMonitor originHealthStatusMonitor, HttpConfig httpConfig,
-                                                  MetricRegistry metricsRegistry, OriginStatsFactory originStatsFactory) {
-            await(originHealthStatusMonitor.start());
-
-            ConnectionPool.Factory hostConnectionPoolFactory = connectionPoolFactory(backendService.connectionPoolConfig(), httpConfig, metricsRegistry, originStatsFactory);
-            OriginsInventory originsInventory = new OriginsInventory(eventBus, backendService.id(), originHealthStatusMonitor, hostConnectionPoolFactory, metricsRegistry);
-            originsInventory.setOrigins(backendService.origins());
+            OriginsInventory originsInventory = new OriginsInventory(eventBus, appId, originHealthMonitor, connectionPoolFactory, metricsRegistry);
+            originsInventory.setOrigins(initialOrigins);
```

Comment:
I think the `setOrigins(Set<Origin>)` method can be private, and `setOrigins(Origin...)` can be package-private.

Reply:
Hi Kyle. The `setOrigins(Set<Origin>)` is nowadays public. This is because the inventory by design will manage the state for the origins, and the `setOrigins` provides a way to push more origins into the inventory.